[receiver/awscloudwatch] Added pattern filter option for log groups (#40981)
open-telemetry/opentelemetry-collector-contrib: receiver/awscloudwatchreceiver/README.md

- `autodiscover`
  - `limit`: (optional; default = 50) Limits the number of discovered log groups. This does not limit how large each API call to discover the log groups will be.
  - `prefix`: (optional) A prefix for log groups to limit the number of log groups discovered.

>     - Only one of `prefix` or `pattern` can be specified. If both are omitted, all log streams up to the limit are collected from.

Reviewer: ```suggestion
    - Only one of `prefix` or `pattern` can be specified. If both are omitted, all log streams up to the limit are collected.
```

>   - `pattern`: (optional) A case-sensitive substring search on the log group names to limit the number of log groups discovered.

Reviewer: 1. I think it's important to note that it's not a regex, as "pattern" implies regex for many people.
2. The value is not a "case sensitive subtring search", that's the operation that is done using the value.

```suggestion
  - `pattern`: (optional) A case-sensitive substring (not a regular expression) that must be present in the log group names, used to limit the number of log groups discovered.
```

>     - Only one of `prefix` or `pattern` can be specified. If both are omitted, all log streams up to the limit are collected from.

Reviewer: ```suggestion
    - Only one of `prefix` or `pattern` can be specified. If both are omitted, all log streams up to the limit are collected.
```
